projects/packages/my-jetpack/_inc/hooks/use-my-jetpack-connection/index.ts

```diff
@@ -0,0 +1,46 @@
+import { useConnection } from '@automattic/jetpack-connection';
+import {
+	getMyJetpackWindowInitialState,
+	getMyJetpackWindowRestState,
+	getMyJetpackWindowConnectionState,
+} from '../../data/utils/get-my-jetpack-window-state';
+/**
+ * React custom hook to get the site purchases data.
+ *
+ * @returns {object} site purchases data
+ */
+
+interface MyJetpackConnection {
+	apiNonce: string;
+	apiRoot: string;
+	blogID: number;
+	registrationNonce: string;
+	isSiteConnected: boolean;
+	topJetpackMenuItemUrl: string;
+	// The useConnection hook is not typed, so we don't know what other properties it returns.
+	// We could define the types here, but that hook returns a lot of data and it's not best practices
+	// to duplicate them here. The best approach would be to type the useConnection hook itself.
+	[ key: string ]: unknown;
```

Comment:
@robertsreberski  I decided here that since we are not refactoring the components that use this to TS yet that it's best not to type the whole `useConnection` hook in this function. Whenever we do get around to fully converting My Jetpack to TypeScript, I am not sure if it would be best to add typing to the `useConnection` hook in the `jetpack-connection` package or if it would be better to type it here and not touch another package 🤔 

Anyway, that is for later, just wanted to let you know my thought process on this

Reply:
That makes sense, looks good!